### PR TITLE
Update and rename events.md to notifications.md

### PR DESCRIPTION
**Pull request by nathanwoulfe**
Adds description for all (I think I got em all) notifications, renames file from Events

### CODE REVIEW

File: umbraco-workflow/notifications.md [hunk 1]
```diff
@@ -0,0 +1,769 @@
+# Using Notifications
+
+Umbraco Workflow uses Notifications to allow you to hook into the processes for the backoffice. For example, you might want to execute some code every time an approval group is updated. Notifications allow you to do that. For more information on how Umbraco implements Notifications, visit [Using Notifications in the CMS documentation](https://docs.umbraco.com/umbraco-cms/reference/notifications)
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:3]:
```suggestion
Umbraco Workflow uses Notifications to allow you to hook into the processes for the backoffice. For example, you might want to execute some code every time an approval group is updated. Notifications allow you to do that. For more information on how Umbraco implements Notifications, see the [Using Notifications](https://docs.umbraco.com/umbraco-cms/reference/notifications) article.
```

File: umbraco-workflow/notifications.md [hunk 2]
```diff
@@ -0,0 +1,769 @@
...
+
+## Notifications
+
+All notifications reside in the `Umbraco.Workflow.Core.Notifications` namespace and are postfixed with `Notification`.
+
+Typically, available notifications exist in pairs, with a "before" and "after" notification. For example, the SettingsService class publishes both a WorkflowSettingsSavingNotification and a WorkflowSettingsSavedNotification when settings are modified.
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:9]:
```suggestion
Typically, available notifications exist in pairs, with a "before" and "after" notification. For example, the `SettingsService` class publishes both a `WorkflowSettingsSavingNotification` and a `WorkflowSettingsSavedNotification` when settings are modified.
```

File: umbraco-workflow/notifications.md [hunk 3]
```diff
@@ -0,0 +1,769 @@
...
+
+Which one you want to use depends on what you want to achieve. If you want to be able to cancel the action, you would use the "before" notification, and use the CancelOperation method on the notification to cancel it. If you want to execute some code after the settings have been updated, then you would use the "after" notification.
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:11]:
```suggestion
Which one you want to use depends on what you want to achieve. If you want to cancel the action, you will use the "before" notification and use the CancelOperation method on the notification to cancel it. If you want to execute some code after the settings have been updated, then you would use the "after" notification.
```

File: umbraco-workflow/notifications.md [hunk 4]
```diff
@@ -0,0 +1,769 @@
...
+    <td>WorkflowContentReviewsConfigSavedNotification</td>
+    <td>
+      <ul>
+        <li>IEnumerable<IWorkflowConfig> SavedEntities</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>
+      </ul>
+    </td>
+    <td>
+    Published when ContentReviewService.SaveContentReviewConfig is called in the API, after the entities have been saved.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:47]:
```suggestion
    Published when ContentReviewService.SaveContentReviewConfig is called in the API after the entities have been saved.<br/>
```

File: umbraco-workflow/notifications.md [hunk 5]
```diff
@@ -0,0 +1,769 @@
...
+    <td>
+      <ul>
+        <li>Dictionary&ltUserGroupPoco, List&ltContentReviewConfigPoco&gt&gt Target</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>        
+        <li>bool Cancel</li>
+      </ul>
+    </td>
+    <td>
+    Published when ContentReviewReminderEmailer.SendReviewReminders is called in the API, before email notifications are sent.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:62]:
```suggestion
    Published when ContentReviewReminderEmailer.SendReviewReminders is called in the API before email notifications are sent.<br/>
```

File: umbraco-workflow/notifications.md [hunk 6]
```diff
@@ -0,0 +1,769 @@
...
+    <td>WorkflowContentReviewsEmailNotificationsSentNotification</td>
+    <td>
+      <ul>
+        <li>Dictionary&ltUserGroupPoco, List&ltContentReviewConfigPoco&gt&gt Target</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>      
+      </ul>
+    </td>
+    <td>
+    Published when ContentReviewReminderEmailer.SendReviewReminders is called in the API, after email notifications are sent.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:76]:
```suggestion
    Published when ContentReviewReminderEmailer.SendReviewReminders is called in the API after email notifications are sent.<br/>    
```

File: umbraco-workflow/notifications.md [hunk 7]
```diff
@@ -0,0 +1,769 @@
...
+    <td>
+      <ul>
+        <li>IEnumerable&ltIWorkflowConfig&gt DeletedEntities</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>
+        <li>bool Cancel</li>
+      </ul>
+    </td>
+    <td>
+    Published when ConfigService.DeleteNodeConfig is called in the API, before the config items are deleted.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:189]:
```suggestion
    Published when ConfigService.DeleteNodeConfig is called in the API before the config items are deleted.<br/>
```

File: umbraco-workflow/notifications.md [hunk 8]
```diff
@@ -0,0 +1,769 @@
...
+      <ul>
+        <li>IEnumerable&ltIWorkflowInstance&gt Target</li>
+        <li>List&EmailTaskListModel&gt Tasks</li>
+        <li>EmailType EmailType</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>        
+      </ul>
+    </td>
+    <td>
+    Published when WorkflowReminderEmailer.Send is called in the API, after email notifications are sent.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:312]:
```suggestion
    Published when WorkflowReminderEmailer.Send is called in the API after email notifications are sent.<br/>
```

File: umbraco-workflow/notifications.md [hunk 9]
```diff
@@ -0,0 +1,769 @@
...
+    Target: Gets the collection of objects describing the workflow instances used to build the email messages<br />
+    Tasks: Gets the collection of objects describing the overdue workflows for each notified user<br />
+    EmailType: Gets the enum value describing the email type, which will always be `EmailType.Reminder`
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:315]:
```suggestion
    EmailType: Gets the enum value describing the email type which will always be `EmailType.Reminder`
```

File: umbraco-workflow/notifications.md [hunk 10]
```diff
@@ -0,0 +1,769 @@
...
+    <td>
+      <ul>
+        <li>IWorkflowGroup SavedEntities</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>        
+        <li>bool Cancel</li>
+      </ul>
+    </td>
+    <td>
+    Published when GroupService.CreateUserGroupAsync is called in the API, before the entity is created.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:341]:
```suggestion
    Published when GroupService.CreateUserGroupAsync is called in the API before the entity is created.<br/>
```

File: umbraco-workflow/notifications.md [hunk 11]
```diff
@@ -0,0 +1,769 @@
...
+    <td>WorkflowGroupCreatedNotification</td>
+    <td>
+      <ul>
+        <li>IWorkflowGroup SavedEntities</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>        
+      </ul>
+    </td>
+    <td>
+    Published when GroupService.CreateUserGroupAsync is called in the API, after the entity has been created.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:355]:
```suggestion
    Published when GroupService.CreateUserGroupAsync is called in the API after the entity has been created.<br/>
```

File: umbraco-workflow/notifications.md [hunk 12]
```diff
@@ -0,0 +1,769 @@
...
+    <td>
+      <ul>
+        <li>IEnumerable&ltIWorkflowGroup&gt DeletedEntities</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>        
+        <li>bool Cancel</li>
+      </ul>
+    </td>
+    <td>
+    Published when GroupService.DeleteUserGroupAsync is called in the API, before the group is deleted.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:370]:
```suggestion
    Published when GroupService.DeleteUserGroupAsync is called in the API before the group is deleted.<br/>
```

File: umbraco-workflow/notifications.md [hunk 13]
```diff
@@ -0,0 +1,769 @@
...
+    <td>WorkflowGroupDeletedNotification</td>
+    <td>
+      <ul>
+        <li>IEnumerable&ltIWorkflowGroup&gt DeletedEntities</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>        
+      </ul>
+    </td>
+    <td>
+    Published when GroupService.DeleteUserGroupAsync is called in the API, after the group is deleted.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:384]:
```suggestion
    Published when GroupService.DeleteUserGroupAsync is called in the API after the group is deleted.<br/>
```

File: umbraco-workflow/notifications.md [hunk 14]
```diff
@@ -0,0 +1,769 @@
...
+    <td>
+      <ul>
+        <li>IEnumerable&ltIWorkflowGroup&gt SavedEntities</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>        
+        <li>bool Cancel</li>
+      </ul>
+    </td>
+    <td>
+    Published when GroupService.UpdateUserGroupAsync is called in the API, before the group is updated.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:399]:
```suggestion
    Published when GroupService.UpdateUserGroupAsync is called in the API before the group is updated.<br/>
```

File: umbraco-workflow/notifications.md [hunk 15]
```diff
@@ -0,0 +1,769 @@
...
+    <td>WorkflowGroupSavedNotification</td>
+    <td>
+      <ul>
+        <li>IEnumerable&ltIWorkflowGroup&gt SavedEntities</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>        
+      </ul>
+    </td>
+    <td>
+    Published when GroupService.UpdateUserGroupAsync is called in the API, after the group is updated.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:413]:
```suggestion
    Published when GroupService.UpdateUserGroupAsync is called in the API after the group is updated.<br/>
```

File: umbraco-workflow/notifications.md [hunk 16]
```diff
@@ -0,0 +1,769 @@
...
+      <ul>
+        <li>IWorkflowInstance Target</li>
+        <li>WorkflowAction Action</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>        
+        <li>bool Cancel</li>
+      </ul>
+    </td>
+    <td>
+    Published when WorkflowProcess.ActionWorkflow is called in the API, before the workflow stage is approved.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:441]:
```suggestion
    Published when WorkflowProcess.ActionWorkflow is called in the API before the workflow stage is approved.<br/>
```

File: umbraco-workflow/notifications.md [hunk 17]
```diff
@@ -0,0 +1,769 @@
...
+      <ul>
+        <li>IWorkflowInstance Target</li>
+        <li>WorkflowAction Action</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>        
+        <li>bool Cancel</li>
+      </ul>
+    </td>
+    <td>
+    Published when WorkflowProcess.ActionWorkflow is called in the API, before the workflow stage is cancelled.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:474]:
```suggestion
    Published when WorkflowProcess.ActionWorkflow is called in the API before the workflow stage is cancelled.<br/>
```

File: umbraco-workflow/notifications.md [hunk 18]
```diff
@@ -0,0 +1,769 @@
...
+    <td>
+      <ul>
+        <li>IWorkflowInstance Target</li>
+        <li>WorkflowAction Action</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>        
+      </ul>
+    </td>
+    <td>
+    Published when WorkflowProcess.ActionWorkflow is called in the API, after the workflow stage is cancelled.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:490]:
```suggestion
    Published when WorkflowProcess.ActionWorkflow is called in the API after the workflow stage is cancelled.<br/>
```

File: umbraco-workflow/notifications.md [hunk 19]
```diff
@@ -0,0 +1,769 @@
...
+    <td>
+      <ul>
+        <li>IWorkflowInstance Target</li>
+        <li>WorkflowType WorkflowType</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>        
+      </ul>
+    </td>
+    <td>
+    Published when WorkflowProcess.HandleCompleteNow<T> or WorkflowProcess.HandleCompleteLater<T> is called in the API, after the workflow is completed.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:506]:
```suggestion
    Published when WorkflowProcess.HandleCompleteNow<T> or WorkflowProcess.HandleCompleteLater<T> is called in the API after the workflow is completed.<br/>
```

File: umbraco-workflow/notifications.md [hunk 20]
```diff
@@ -0,0 +1,769 @@
...
+    <td>
+      <ul>
+        <li>IWorkflowInstance CreatedEntity</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>        
+        <li>bool Cancel</li>
+      </ul>
+    </td>
+    <td>
+    Published when WorkflowProcess.InitiateWorkflow is called in the API, before the workflow is initiated.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:522]:
```suggestion
    Published when WorkflowProcess.InitiateWorkflow is called in the API before the workflow is initiated.<br/>
```

File: umbraco-workflow/notifications.md [hunk 21]
```diff
@@ -0,0 +1,769 @@
...
+    <td>WorkflowTaskCreatedNotification</td>
+    <td>
+      <ul>
+        <li>IWorkflowInstance CreatedEntity</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>        
+      </ul>
+    </td>
+    <td>
+    Published when WorkflowTaskManager.CreateApprovalTask is called in the API, after the workflow task is persisted.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:695]:
```suggestion
    Published when WorkflowTaskManager.CreateApprovalTask is called in the API after the workflow task is persisted.<br/>
```

File: umbraco-workflow/notifications.md [hunk 22]
```diff
@@ -0,0 +1,769 @@
...
+    <td>
+      <ul>
+        <li>IWorkflowInstance Target</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>        
+        <li>bool Cancel</li>
+      </ul>
+    </td>
+    <td>
+    Published when WorkflowTaskManager.ResubmitWorkflow is called in the API, before the workflow task is updated.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:710]:
```suggestion
    Published when WorkflowTaskManager.ResubmitWorkflow is called in the API before the workflow task is updated.<br/>
```

File: umbraco-workflow/notifications.md [hunk 23]
```diff
@@ -0,0 +1,769 @@
...
+    <td>WorkflowTaskUpdatedNotification</td>
+    <td>
+      <ul>
+        <li>IWorkflowInstance Target</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>        
+      </ul>
+    </td>
+    <td>
+    Published when WorkflowTaskManager.ResubmitWorkflow is called in the API, after the workflow task is updated.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:724]:
```suggestion
    Published when WorkflowTaskManager.ResubmitWorkflow is called in the API after the workflow task is updated.<br/>
```

File: umbraco-workflow/notifications.md [hunk 24]
```diff
@@ -0,0 +1,769 @@
...
+    <td>
+      <ul>
+        <li>IEnumerable&ltISettings&gt SavedEntities</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>        
+        <li>bool Cancel</li>
+      </ul>
+    </td>
+    <td>
+    Published when SettingsService.UpdateSettings<T> is called in the API, before the settings are saved.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:751]:
```suggestion
    Published when SettingsService.UpdateSettings<T> is called in the API before the settings are saved.<br/>
```

File: umbraco-workflow/notifications.md [hunk 25]
```diff
@@ -0,0 +1,769 @@
...
+    <td>WorkflowSettingsSavedNotification</td>
+    <td>
+      <ul>
+        <li>IEnumerable&ltISettings&gt SavedEntities</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>        
+      </ul>
+    </td>
+    <td>
+    Published when SettingsService.UpdateSettings<T is called in the API, after the settings are saved.<br/>
```
Comment (eshanrnh) [inline umbraco-workflow/notifications.md:765]:
```suggestion
    Published when SettingsService.UpdateSettings<T is called in the API after the settings are saved.<br/>
```

**Review by eshanrnh — COMMENTED** [review]
Looks good, @nathanwoulfe 🙌 Just a couple of suggestions to fix a vale warning and remove comma.

File: umbraco-workflow/notifications.md [hunk 26]
```diff
@@ -0,0 +1,769 @@
...
+  </tr>
+
+  <tr>
+    <td>WorkflowContentReviewsConfigSavingNotification</td>
+    <td>
+      <ul>
+        <li>IEnumerable<IWorkflowConfig> SavedEntities</li>
+        <li>EventMessages Messages</li>
+        <li>IDictionary&ltstring,object&gt State</li>
+        <li>bool Cancel</li>
```
Comment (nathanwoulfe) [inline umbraco-workflow/notifications.md:29]: Hey @eshanrnh we're getting value warnings for this line, but it's correct for `bool` to be lowercase, so will leave as-is. It's only a warning after all 😄

Comment (eshanrnh) [reply umbraco-workflow/notifications.md:29]: Yup, I agree 😀 That's why I ignored that warning 😉

**Pull request merged**